server/controllers/finance/accounts/references.js

```diff
@@ -257,31 +261,27 @@ function getValue(req, res, next) {
  * @param {Number} id - the id of the account reference
  * @returns {Promise} - a promise resolving to the result of the database.
  */
-function lookupAccountReference(id) {
-  let glb = {};
+async function lookupAccountReference(id) {
   const sql = `
-    SELECT id, abbr, description, parent, is_amo_dep, reference_type_id FROM account_reference WHERE id = ?;`;
+    SELECT id, abbr, description, parent, is_amo_dep, reference_type_id
+    FROM account_reference WHERE id = ?;
+  `;
 
   const sqlItems = `
     SELECT account_id FROM account_reference_item WHERE account_reference_id = ? AND is_exception = 0;`;
 
   const sqlExceptItems = `
     SELECT account_id FROM account_reference_item WHERE account_reference_id = ? AND is_exception = 1;`;
 
-  return db.one(sql, id)
-    .then(reference => {
-      glb = reference;
-      return db.exec(sqlItems, [id]);
-    })
-    .then(referenceItems => {
-      glb.accounts = referenceItems.map(i => i.account_id);
-      return db.exec(sqlExceptItems, [id]);
-    })
-    .then(referenceItems => {
-      glb.accountsException = referenceItems.map(i => i.account_id);
-
-      return glb;
-    });
+  const reference = await db.one(sql, id);
+
+  const [accounts, accountsException] = await Promise.all([
+    db.exec(sqlItems, [id]),
+    db.exec(sqlExceptItems, [id]),
+  ]);
+
+  Object.assign(reference, { accounts, accountsException });
```

Comment:
`accounts` and `accountsException` must be arrays of integer, not arrays of objects `{ account_id : ... }`.

we must have something like : 
```js
reference.accounts = accounts.map(item => item.account_id);
reference.accountsException = accountsException.map(item => item.account_id);
```